sql/util_internal_table_ddl.sql

```diff
@@ -94,7 +94,7 @@ BEGIN
     IF dimension_row.partitioning_func IS NOT NULL THEN
         return format(
             $$
-                %1$I.%2$s(%3$I::text) >= %4$L AND %1$I.%2$s(%3$I::text) <  %5$L
+                %1$I.%2$s(%3$I::text) >= %4$L::bigint AND %1$I.%2$s(%3$I::text) <  %5$L::bigint
```

Comment:
instead of changing the ranges to BIGINT we can make the top range "<=" instead of "<". What do you think is better?

Reply:
I think the best solution is for the start and end slices to have open-ended ranges.

So, the first partition's check constraint would be  ` CHECK (column < value)` and the last partition would be `CHECK (column >= value)`. Only middle slices would have `CHECK(column >= value AND column < value)`.

Reply:
Ideally, this would be the same for the first and last (possible) slices in time dimensions.

Reply:
Why not having both ranges inclusive on all slices ? `>=` and `<=` on all slices ? I see some value in treating all slices the same although it is mainly a gut feeling.